File: rb/features.rb
# UvIndexApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module UvIndexApi2Features
  def self.make_feature(name)
    case name
    when "base"
      UvIndexApi2BaseFeature.new
    when "test"
      UvIndexApi2TestFeature.new
    else
      UvIndexApi2BaseFeature.new
    end
  end
end

File: py/features.py
# UvIndexApi2 SDK feature factory

from feature.base_feature import UvIndexApi2BaseFeature
from feature.test_feature import UvIndexApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: UvIndexApi2BaseFeature(),
        "test": lambda: UvIndexApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
